### PR TITLE
tool: add mono to allow users to run .NET application across various platforms.

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/operating_system.py [hunk 1]
```diff
@@ -487,6 +487,9 @@ def add_azure_core_repo(
     def set_kill_user_processes(self) -> None:
         raise NotImplementedError("set_kill_user_processes is not implemented")
 
+    def add_key(self, server_name: str, key: str) -> None:
```
Comment (squirrelsc) [inline lisa/operating_system.py]: Is it a general method for all distro?

File: lisa/tools/mono.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        required_packages = [
+            "dirmngr",
+            "gnupg",
+            "apt-transport-https",
+            "ca-certificates",
+        ]
+        posix_os.install_packages(required_packages)
+
+        mono_repo_entry = (
+            "deb https://download.mono-project.com/repo/ubuntu stable-bionic main"
```
Comment (squirrelsc) [inline lisa/tools/mono.py]: Does it support only bionic? If so, filter out other versions.

File: lisa/tools/mono.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            value=mono_repo_entry,
+            file=self.node.get_pure_path(
+                "/etc/apt/sources.list.d/mono-official-stable.list"
+            ),
+            sudo=True,
+        )
+
+        posix_os.add_key(
+            server_name="hkp://keyserver.ubuntu.com:80",
+            key="3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF",
```
Comment (squirrelsc) [inline lisa/tools/mono.py]: Where is it from? Is it public?

File: lisa/operating_system.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: I know for fedora, SUSE, they use `rpm --import key`,

File: lisa/tools/mono.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/tools/mono.py]: add comments, it is public

File: lisa/tools/mono.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/tools/mono.py]: change into distro codename

File: lisa/operating_system.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/operating_system.py]: Let's keep it in Debian only, until we implement another os, and find the common pattern.

File: lisa/tools/mono.py (hunk 3 above)
Comment (squirrelsc) [reply lisa/tools/mono.py]: Please also provide the source document.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**